addons/sale/models/sale_order_line.py

```diff
@@ -1395,7 +1395,6 @@ def _prepare_invoice_line(self, **optional_values):
             'sale_line_ids': [Command.link(self.id)],
             'is_downpayment': self.is_downpayment,
         }
-        self._set_analytic_distribution(res, **optional_values)
```

Comment:
Shouldn't this be done in `pos_sale` as well, where `_set_analytic_distribution` is called?